### PR TITLE
Move user filters to filter table

### PR DESCRIPTION
**Pull request by Beacze**
This pull request removes overhead of storing user filters inside the property table.

- flyway migration
- hibernate mapping
- bean class
- DAO class
- type class
- service class
- adjustment of dependent classes
- tests

### CODE REVIEW

**Comment by henning-gerhardt** [comment]
Travis CI reports compile error:
```
[ERROR] /home/travis/build/kitodo/kitodo-production/Kitodo/src/test/java/org/kitodo/services/data/FilterServiceIT.java:[35,40] error: unreported exception DataException; must be caught or declared to be thrown
```

File: Kitodo-DataManagement/src/main/resources/db/migration/V2_9__Move_user_property_to_filter_table.sql [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+-- 7. Add foreign keys to user table
+
+ALTER TABLE filter ENGINE=InnoDB;
+ALTER TABLE filter
+  ADD CONSTRAINT `FK_filter_x_user_id`
+FOREIGN KEY (user_id) REFERENCES user (id);
+
+-- 8. Delete all filters from property table
+
+DELETE FROM property WHERE title = '_filter';
```
Comment (Kathrin-Huber) [inline Kitodo-DataManagement/src/main/resources/db/migration/V2_9__Move_user_property_to_filter_table.sql]: I like that script, it worked like a charm :)

File: Kitodo-DataManagement/src/main/java/org/kitodo/data/database/persistence/FilterDAO.java [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+     * @param query
+     *            as String
+     * @return list of properties
+     */
+    @SuppressWarnings("unchecked")
+    public List<Filter> search(String query) throws DAOException {
+        return retrieveObjects(query);
+    }
+
+    public Filter save(Filter filter) throws DAOException {
```
Comment (solth) [inline Kitodo-DataManagement/src/main/java/org/kitodo/data/database/persistence/FilterDAO.java]: JavaDoc missing

Comment (Beacze) [inline Kitodo-DataManagement/src/main/java/org/kitodo/data/database/persistence/FilterDAO.java]: added

**Pull request merged**